### ISSUE TITLE
Improved Import of Multiple Resources

### ISSUE THREAD

**Issue by paulaustin-automutatio** [opentofu/opentofu#1282]
### OpenTofu Version

```shell
OpenTofu v1.6.1
on windows_amd64
```

### Use Cases

If the state becomes corrupted you would have to import many existing resources into a new state.

At the moment each import is run as a single command which has  a lot of overhead.

### Attempted Solutions

PowerShell (or bash) scripts that creates an array of resources to import, then looping through tofu import for each resource.

### Proposal

Here are several potential proposals.

1. Have a command line parameter (e.g. --import-file imports.(json|csv)) so specify a json or csv file with the resources to import

Then the import command loads the configuration only once and loops through all the resources to import (maybe in several threads).

2. This one is a bit more complicated. Have the ability for a providers implement an auto-import function for data and resources. On tofu import have a flag --auto-import that would get the resource ids from the provider and then display the list of proposed imports.

For example in the azurerm provider the values of the the import ids can be calculated from the provider subscription id and the paremeters in the resource.

The information is in the documentation, I haven't looked to see if the docs are generated from metadata in the provider code.

/subscriptions/**00000000-0000-0000-0000-000000000000**/resourceGroups/**mygroup1**/providers/Microsoft.Network/virtualNetworks/**myvnet1**

**Comment by janosdebugs** [comment]
Hi @paulaustin-automutatio thank you for the issue. Have you tried using the [import block](https://opentofu.org/docs/language/import/)?

**Comment by paulaustin-automutatio** [comment]
@janosdebugs thanks for the quick reply. I completely missed that the import blocks were added in 1.5.

Much better solution than my first proposal.

**Comment by paulaustin-automutatio** [comment]
@janosdebugs A couple of comments for the future.

1. The import block doesn't seem to support for_each, which would be useful where the resources were created with for_each
2. An almost simpler approach would be to add support in the resource definition an import_id property (or tofu_import_id if concerned about providers having a import_id propertyin their resources)

**Comment by paulaustin-automutatio** [comment]
Added new issue https://github.com/opentofu/opentofu/issues/1287
